robowflex_library/scripts/cob4_visualization.cpp

```diff
@@ -0,0 +1,166 @@
+    scene->fromYAMLFile("package://robowflex_library/yaml/test_cob4.yml");
+
+    // Visualize the scene.
+    rviz.visualizeCurrentState();
+    rviz.updateScene(scene);
+
+    // Create the default planner for the COB4.
+    auto planner = std::make_shared<OMPL::OMPLPipelinePlanner>(cob4);
+    // Path simplification disabled.
+    OMPL::Settings settings;
+    //    settings.simplify_solutions = false;
+    planner->initialize("package://robowflex_resources/cob/config/ompl_planning.yaml",  // planner config
+                        settings);
+
+    // Sets the Cob4's base pose.
+    cob4->pointHead({0.6, -0.26, 0.95});
+
+    // Create a motion planning request with a joint position goal for the right arm.
+    MotionRequestBuilder request_right_arm(planner, RIGHT_ARM);
+    request_right_arm.setStartConfiguration(cob4->getScratchState());
+
+    // Create a motion planning request with a pose goal for the right arm.
+    RobotPose goal_pose_right = RobotPose::Identity();
+    goal_pose_right.translate(Eigen::Vector3d{0.6, -0.26, 0.95});
+    goal_pose_right.rotate(Eigen::Quaterniond{0.0, 1.0, 0.0, 0.0});
+    request_right_arm.setGoalPose(RIGHT_EE, "base_link", goal_pose_right);
+
+    // Visualizing the target poses for the right arm.
+    rviz.addGoalMarker("goal", request_right_arm);
+    rviz.addTransformMarker("pose_right", "base_link", goal_pose_right);
+    rviz.updateMarkers();
+
+    RBX_INFO("Scene and Goal displayed for the right arm! Press enter to plan...");
+    std::cin.get();
+
+    // Do motion planning!
+    request_right_arm.setConfig("RRTConnect");
+    planning_interface::MotionPlanResponse res = planner->plan(scene, request_right_arm.getRequest());
+    if (res.error_code_.val != moveit_msgs::MoveItErrorCodes::SUCCESS)
+        return 1;
+
+    // Publish the trajectory to a topic to display in RViz
+    rviz.updateTrajectory(res);
+
+    // Visualize resulting state.
+    trajectory_msgs::JointTrajectoryPoint point;
+    res.trajectory_->getLastWayPoint().copyJointGroupPositions(
+        res.trajectory_->getLastWayPoint().getJointModelGroup(RIGHT_ARM), point.positions);
+    cob4->setGroupState(RIGHT_ARM, std::vector<double>(point.positions));
+
+    rviz.visualizeCurrentState();
+
+    RBX_INFO("Press enter to continue with the left arm.");
+
+    // Sets the Cob4's base pose.
+    cob4->pointHead({0.4, 0.26, 0.79});
+
+    // Create a motion planning request with a joint position goal for the right arm.
+    MotionRequestBuilder request_left_arm(planner, LEFT_ARM);
+    request_left_arm.setStartConfiguration(cob4->getScratchState());
+
+    // Create a motion planning request with a pose goal for the left arm.
+    RobotPose goal_pose_left = RobotPose::Identity();
+    goal_pose_left.translate(Eigen::Vector3d{0.38, 0.26, 0.76});
+    goal_pose_left.rotate(Eigen::Quaterniond{0.707, 0.0, 0.707, 0.0});
+    request_left_arm.setGoalPose(LEFT_EE, "base_link", goal_pose_left);
```

Comment:
Use the TF::createPose* functions to clean this up.

Reply:
Done!